File: sirius-core/src/main/java/org/starcoin/io/IOUtils.java
package org.starcoin.io;

import com.google.common.io.ByteStreams;

import java.io.Closeable;
import java.io.IOException;
import java.io.InputStream;
import java.nio.ByteBuffer;

public class IOUtils {

    public static ByteBuffer toByteBuffer(InputStream in) throws IOException {
        return ByteBuffer.wrap(ByteStreams.toByteArray(in));
    }

    public static void close(Closeable closeable) {
        if (closeable == null) {
            return;
        }
        try {
            closeable.close();
        } catch (IOException e) {
            // just ignore
        }
    }
}

File: sirius-core/src/main/kotlin/org/starcoin/sirius/io/IOUtils.kt
package org.starcoin.sirius.io

import com.google.common.io.ByteStreams

import java.io.Closeable
import java.io.IOException
import java.io.InputStream
import java.nio.ByteBuffer

object IOUtils {

    @Throws(IOException::class)
    fun toByteBuffer(`in`: InputStream): ByteBuffer {
        return ByteBuffer.wrap(ByteStreams.toByteArray(`in`))
    }

    fun close(closeable: Closeable?) {
        if (closeable == null) {
            return
        }
        try {
            closeable.close()
        } catch (e: IOException) {
            // just ignore
        }

    }
}
